### PR TITLE
Support generating types for ES Modules

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

**Comment by rictic** [comment]
I've finally got a working compile locally so I can add some tests. Will add them in tomorrow.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -156,6 +157,9 @@
 
   private static final Splitter DOT_SPLITTER = Splitter.on('.');
 
+  private static final ImmutableList<String> PACKAGES_TO_IGNORE =
+      ImmutableList.of("third_party/javascript/polymer/v2");
```
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Would be better for this to be a externally passed in flag. Also, needs more docs.

**Comment by rkirov** [comment]
Looks pretty straightforward at a glance. Let me know when the tests are ready to do a proper code review.

File: src/main/java/com/google/javascript/clutz/Options.java [hunk 2]
```diff
@@ -202,6 +204,11 @@ public CompilerOptions getCompilerOptions() {
     // Report duplicate definitions, e.g. for accidentally duplicated externs.
     options.setWarningLevel(DiagnosticGroups.DUPLICATE_VARS, CheckLevel.ERROR);
 
+    // Handle modules in the same way the production compiler does by default.
+    // See com/google/javascript/jscomp/FlagBasedOptions.java
+    options.setModuleResolutionMode(ResolutionMode.BROWSER_WITH_TRANSFORMED_PREFIXES);
+    options.setBrowserResolverPrefixReplacements(ImmutableMap.of("google3/", "/"));
```
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/Options.java]: I am handling this part with #848

**Comment by rkirov** [comment]
We are moving to g3 first style of development for clutz, so you can redo this internally first, if it is still relevant.

**Pull request closed without merging**